From a920e94a9cf7720d2c38a5a126117c34714ca8d3 Mon Sep 17 00:00:00 2001
From: Alexey Prokhin <alexey@prokhin.ru>
Date: Wed, 17 Oct 2018 19:55:27 +0300
Subject: [PATCH] winex11.drv: Ignore clip_reset when trying to clip the mouse
 after the desktop has been resized.

This fixes the mouse clipping when the desktop is resized multiple times in a row.

CW-Bug-Id: #21879
---
 dlls/win32u/input.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dlls/win32u/input.c b/dlls/win32u/input.c
index 08465270154..9aed08b1a20 100644
--- a/dlls/win32u/input.c
+++ b/dlls/win32u/input.c
@@ -2811,7 +2811,7 @@ BOOL clip_fullscreen_window( HWND hwnd, BOOL reset )
 
     if (!ret || !is_fullscreen( &monitor_info, &window_rect )) return FALSE;
     if (is_captured_by_system()) return FALSE;
-    if (NtGetTickCount() - thread_info->clipping_reset < 1000) return FALSE;
+    if (!reset && NtGetTickCount() - thread_info->clipping_reset < 1000) return FALSE;
     if (!reset && clipping_cursor && thread_info->clipping_cursor) return FALSE;  /* already clipping */
 
     if (!grab_fullscreen)
-- 
2.56.0

